fdbserver/KeyValueStoreShardedRocksDB.actor.cpp

```diff
@@ -489,22 +502,153 @@ struct PhysicalShard {
+	options.deadline = std::chrono::duration_cast<std::chrono::microseconds>(deadlineSeconds);
+
+	// When using a prefix extractor, ensure that keys are returned in order even if they cross
+	// a prefix boundary.
+	options.auto_prefix_mode = (SERVER_KNOBS->ROCKSDB_PREFIX_LEN > 0);
+	if (rowLimit >= 0) {
+		ReadIterator readIter = shard->physicalShard->readIterPool->getIterator();
+		auto cursor = readIter.iter;
+		cursor->Seek(toSlice(range.begin));
+		while (cursor->Valid() && toStringRef(cursor->key()) < range.end) {
+			KeyValueRef kv(toStringRef(cursor->key()), toStringRef(cursor->value()));
+			++accumulatedRows;
+			accumulatedBytes += sizeof(KeyValueRef) + kv.expectedSize();
+			result->push_back_deep(result->arena(), kv);
+			// Calling `cursor->Next()` is potentially expensive, so short-circut here just in case.
+			if (result->size() >= rowLimit || accumulatedBytes >= byteLimit) {
+				break;
+			}
+			cursor->Next();
+		}
+		s = cursor->status();
+		shard->physicalShard->readIterPool->returnIterator(readIter);
+	} else {
+		ReadIterator readIter = shard->physicalShard->readIterPool->getIterator();
+		auto cursor = readIter.iter;
+		cursor->SeekForPrev(toSlice(range.end));
+		if (cursor->Valid() && toStringRef(cursor->key()) == range.end) {
+			cursor->Prev();
+		}
+		while (cursor->Valid() && toStringRef(cursor->key()) >= range.begin) {
+			KeyValueRef kv(toStringRef(cursor->key()), toStringRef(cursor->value()));
+			++accumulatedRows;
+			accumulatedBytes += sizeof(KeyValueRef) + kv.expectedSize();
+			result->push_back_deep(result->arena(), kv);
+			// Calling `cursor->Prev()` is potentially expensive, so short-circut here just in case.
+			if (result->size() >= -rowLimit || accumulatedBytes >= byteLimit) {
+				break;
+			}
+			cursor->Prev();
+		}
+		s = cursor->status();
+		shard->physicalShard->readIterPool->returnIterator(readIter);
+	}
+
+	if (!s.ok()) {
+		logRocksDBError(s, "ReadRange");
+		// The data writen to the arena is not erased, which will leave RangeResult in a dirty state. The RangeResult
+		// should never be returned to user.
+		return -1;
+	}
+	return accumulatedBytes;
+}
+
 // Manages physical shards and maintains logical shard mapping.
 class ShardManager {
 public:
 	ShardManager(std::string path) : path(path) {}
 	rocksdb::Status init() {
-		// TODO: Open default cf and get shard mapping.
+		dataShardMap.insert(allKeys, nullptr);
 		// Open instance.
-		auto options = getOptions();
-		rocksdb::Status status = rocksdb::DB::Open(options, path, &db);
+		std::vector<std::string> columnFamilies;
+		rocksdb::Options options = getOptions();
+		rocksdb::Status status = rocksdb::DB::ListColumnFamilies(options, path, &columnFamilies);
+
+		rocksdb::ColumnFamilyOptions cfOptions = getCFOptions();
+		std::vector<rocksdb::ColumnFamilyDescriptor> descriptors;
+		bool foundMetadata = false;
+		for (const auto& name : columnFamilies) {
+			if (name == "kvs-metadata") {
```

Comment:
If we don't use the `default` column family at all, the it will be an avoidable waste, even the waste is small. 
Can we use a global `const std::string kvsMetaData = "default";` to address the naming issue?